presto-spark-base/src/main/java/com/facebook/presto/spark/execution/LocalTask.java

```diff
@@ -0,0 +1,261 @@
+            this.taskUpdateRequestCodec = taskUpdateRequestCodec;
+            this.planFragmentCodec = planFragmentCodec;
+            this.taskInfoCodec = taskInfoCodec;
+        }
+    }
+
+    public TaskInfo getTaskInfo()
+    {
+        return taskInfo;
+    }
+
+    public void start()
+    {
+        sendUpdate();
+        scheduleGetTaskInfo();
+    }
+    private synchronized void sendUpdate()
+    {
+        Optional<byte[]> fragment = Optional.of(planFragment.toBytes(planFragmentCodec));
+        Optional<TableWriteInfo> writeInfo = Optional.of(tableWriteInfo);
+        TaskUpdateRequest updateRequest = new TaskUpdateRequest(
+                session.toSessionRepresentation(),
+                session.getIdentity().getExtraCredentials(),
+                fragment,
+                sources,
+                outputBuffers.get(),
+                writeInfo);
+        byte[] taskUpdateRequestJson = taskUpdateRequestCodec.toBytes(updateRequest);
+
+        HttpUriBuilder uriBuilder = uriBuilderFrom(taskInfo.getTaskStatus().getSelf());
+        Request request = setContentTypeHeaders(false, preparePost())
+                .setUri(uriBuilder.build())
+                .setBodyGenerator(createStaticBodyGenerator(taskUpdateRequestJson))
+                .build();
+
+        ResponseHandler responseHandler = createAdaptingJsonResponseHandler(taskInfoCodec);
+        ListenableFuture<BaseResponse<TaskInfo>> future = httpClient.executeAsync(request, responseHandler);
+        FutureCallback callback = new SimpleHttpResponseHandler<>(new UpdateResponseHandler(), request.getUri(), stats.getHttpResponseStats(), LOCAL_TASK_ERROR);
+
+        Futures.addCallback(
+                future,
+                callback,
+                executor);
+    }
+
+    private synchronized void scheduleGetTaskInfo()
+    {
+        scheduledFuture = updateScheduledExecutor.scheduleWithFixedDelay(() -> {
+            try {
+                HttpUriBuilder uriBuilder = uriBuilderFrom(taskInfo.getTaskStatus().getSelf());
+                Request request = setContentTypeHeaders(false, prepareGet())
+                        .setUri(uriBuilder.build())
+                        .build();
+
+                ResponseHandler responseHandler = createAdaptingJsonResponseHandler(taskInfoCodec);
+                ListenableFuture<BaseResponse<TaskInfo>> future = httpClient.executeAsync(request, responseHandler);
+                FutureCallback callback = new SimpleHttpResponseHandler<>(new TaskInfoHandler(), request.getUri(), stats.getHttpResponseStats(), LOCAL_TASK_ERROR);
+
+                Futures.addCallback(
+                        future,
+                        callback,
+                        executor);
+            }
+            catch (Throwable t) {
+                throw t;
+            }
+        }, 0, 100, MILLISECONDS);
+    }
+
+    public synchronized void stop()
+    {
+        if (scheduledFuture != null) {
+            scheduledFuture.cancel(true);
```

Comment:
I don't know too much about this but I passed false as I think it might be safer. Do we have a preference?